corehq/apps/locations/permissions.py

```diff
@@ -230,13 +228,16 @@ def _inner(view_fn):
 def location_restricted_response(request):
     from corehq.apps.hqwebapp.views import no_permissions
     notify_exception(request, NOTIFY_EXCEPTION_MSG)
-    return no_permissions(request, message=LOCATION_ACCESS_DENIED)
+    return no_permissions(request, exception=LocationPermissionDenied())
 
 
 def location_restricted_exception(request):
-    from corehq.apps.hqwebapp.views import no_permissions_exception
     notify_exception(request, NOTIFY_EXCEPTION_MSG)
-    return no_permissions_exception(request, message=LOCATION_ACCESS_DENIED)
+    return LocationPermissionDenied()
```

Comment:
Raising a 403 automatically redirects to the `/403` endpoint right? But like this, the `message` block won't be populated?